### ISSUE TITLE
macro qualification inconsistency

### ISSUE THREAD

**Issue by StefanKarpinski** [JuliaLang/julia#1769]
The way one qualifies macro names seems kind of inconsistent:

``` julia
import Mod.@foo  # macro import
@Mod.foo         # invocation of unimported macro
```

Is the `@` part of the name or the invocation? Seems like right now it's kind of both which is a bit weird. Tangentially, something I've long wanted is a short form for defining macros along the lines of this:

``` julia
@foo(ex) = f(ex)
```

**Comment by ihnorton** [comment]
Is the un-imported invocation syntax still necessary? It's only used in one test in base. Of the ~130 packages I have installed, it is used sparingly in: JuMP and MPB, PyCall, FactCheck, ASTInterpreter, ELF, Calculus, BinDeps, TerminalUI, and TexExtensions (mostly once each, except for ~10 uses in the JuMP tests).

(x-ref [-users post](https://groups.google.com/d/msg/julia-users/KOc2DrPbaWM/58ZKgq_cAgAJ), #1107)

**Comment by yuyichao** [comment]
IMHO allowing both `@M.foo` and `M.@foo` doesn't seem to be a big problem.

**Comment by tkelman** [comment]
You need some way of being able to call macros of the same name from different packages. I think `M.@foo` is a bit more consistent and we could move to deprecate `@M.foo` if we really felt the need.

**Comment by JeffreySarnoff** [comment]
do it

**Comment by StefanKarpinski** [comment]
It's too late now, we're stuck with both syntaxes until 2.0.

**Comment by ararslan** [comment]
Is there an open issue to track deprecating `@M.foo` for 2.0?

**Comment by StefanKarpinski** [comment]
No, I guess we can reopen this and add it to the 2.0 milestone.

**Comment by Keno** [comment]
I've said this elsewhere (but I guess this is the right place for it. I actually kind of like the `@` at the start because it makes the evaluation scope of the expression clear. In particular, without cheating, can you tell me in what scope the following runs:
```
foo() = (println("Hello"); Base).Experimental.@VERSION
```
